File: frontend/src/services/commentService.ts
import { apiClient } from '../api/client';
import { ApiResponse } from '../types';

export interface CommentResponse {
  id: number;
  ticketId: number;
  userId: number;
  userName: string;
  userRole: string;
  content: string;
  createdAt: string;
}

export interface CreateCommentRequest {
  ticketId: number;
  content: string;
}

export const commentService = {
  addComment: async (data: CreateCommentRequest): Promise<ApiResponse<CommentResponse>> => {
    const response = await apiClient.post<ApiResponse<CommentResponse>>('/api/v1/comments', data);
    return response.data;
  },

  getCommentsByTicket: async (ticketId: number): Promise<ApiResponse<CommentResponse[]>> => {
    const response = await apiClient.get<ApiResponse<CommentResponse[]>>(`/api/v1/comments/ticket/${ticketId}`);
    return response.data;
  },
};

File: backend/src/main/java/com/csms/comment/service/CommentService.java
package com.csms.comment.service;

import com.csms.common.response.ApiResponse;
import com.csms.comment.dto.CommentResponse;
import com.csms.comment.dto.CreateCommentRequest;

import java.util.List;

public interface CommentService {
    ApiResponse<CommentResponse> addComment(CreateCommentRequest request, String userEmail);
    ApiResponse<List<CommentResponse>> getCommentsByTicket(Long ticketId, String userEmail);
}
